src/Sns/Input/PublishInput.php

```diff
@@ -134,41 +134,34 @@ public function requestBody(): string
         if (null !== $v = $this->TopicArn) {
             $payload['TopicArn'] = $v;
         }
-
         if (null !== $v = $this->TargetArn) {
             $payload['TargetArn'] = $v;
         }
-
         if (null !== $v = $this->PhoneNumber) {
             $payload['PhoneNumber'] = $v;
         }
-
         $payload['Message'] = $this->Message;
-
         if (null !== $v = $this->Subject) {
             $payload['Subject'] = $v;
         }
-
         if (null !== $v = $this->MessageStructure) {
             $payload['MessageStructure'] = $v;
         }
 
         (static function (array $input) use (&$payload, $indices) {
-            $indices->kb0b4646 = 0;
+            $indices->kd0379ae = 0;
             foreach ($input as $key => $value) {
-                ++$indices->kb0b4646;
-                $payload["MessageAttributes.{$indices->kb0b4646}.Name"] = $key;
+                ++$indices->kd0379ae;
+                $payload["MessageAttributes.entry.{$indices->kd0379ae}.Name"] = $key;
```

Comment:
List contains a `member` key, and map contains a `entry` key (see https://docs.aws.amazon.com/fr_fr/sns/latest/api/API_Publish.html)

Reply:
Great catch!